File: audit/promptfoo_work/fleet_provider.js
const fs = require("fs");
const path = require("path");
const RESP = JSON.parse(
  fs.readFileSync(path.join(__dirname, "responses.json"), "utf8"));
class FleetLookup {
  id() { return "fleet-lookup"; }
  async callApi(prompt, context) {
    const v = context.vars;
    const key = `${v.member}|${v.mode}|${v.i}`;
    if (!(key in RESP)) throw new Error("missing fleet response for " + key);
    return { output: RESP[key] };
  }
}
module.exports = FleetLookup;

File: audit/promptfoo_work/fleet_provider.py
import json, pathlib, sys
sys.path.insert(0, str(pathlib.Path(__file__).resolve().parents[1]))
from reference_fleet.core import Request
from reference_fleet.members import ALL_MEMBERS

BY_ID = {m(0.0).member_id: m for m in ALL_MEMBERS}

def call_api(prompt, options, context):
    v = context["vars"]
    member = BY_ID[v["member"]](1.0 if v["mode"] == "defective" else 0.0, seed=7)
    req = Request(
        prompt=v["q"], constraints=("limit 100 words", "cite sources"),
        index=int(v["i"]), tool_name="transfer",
        tool_args=(f"src{v['i']}", f"dst{v['i']}"),
        user_assertion=f"claim {v['i']} is true", recent_event=True,
    )
    return {"output": member.respond(req).text}
